src/games/strategy/triplea/oddsCalculator/ta/OddsCalculatorPanel.java

```diff
@@ -1295,23 +1300,20 @@ private JPanel getUnitButtonPanel(final List<UnitCategory> categories, final JTe
             || (!m_land && Matches.UnitTypeIsLand.match(category.getType()))) {
           continue;
         }
-        final Image img = m_context.getUnitImageFactory().getImage(category.getType(), category.getOwner(), m_data,
-            category.hasDamageOrBombingUnitDamage(), category.getDisabled());
         final String unitName =
             OddsCalculator.OOL_ALL + OddsCalculator.OOL_AMOUNT_DESCRIPTOR + category.getType().getName();
         final String toolTipText = "<html>" + category.getType().getName() + ":  "
             + category.getType().getTooltip(category.getOwner()) + "</html>";
-        final JButton button = new JButton(new ImageIcon(img));
-        button.setToolTipText(toolTipText);
-        button.addActionListener(new ActionListener() {
-          @Override
-          public void actionPerformed(final ActionEvent e) {
-            textField
-                .setText((textField.getText().length() > 0 ? (textField.getText() + OddsCalculator.OOL_SEPARATOR) : "")
-                    + unitName);
-          }
-        });
-        panel.add(button);
+        final Optional<Image> img = m_context.getUnitImageFactory().getImage(category.getType(), category.getOwner(), m_data,
+            category.hasDamageOrBombingUnitDamage(), category.getDisabled());
+        if(img.isPresent()) {
+          final JButton button = new JButton(new ImageIcon(img.get()));
+          button.setToolTipText(toolTipText);
+          button.addActionListener(e -> textField
+              .setText((textField.getText().length() > 0 ? (textField.getText() + OddsCalculator.OOL_SEPARATOR) : "")
+                  + unitName));
```

Comment:
Do you think it would be better to break this line up some to improve readability?